### ISSUE TITLE
🧑‍💻:  Upgrade Project: Add Streamlit Web App, Lasso Regression, and Improve Documentation

### ISSUE THREAD

**Issue by Shivu384** [UTSAVS26/PyVerse#1402]
## 📌 Description

This issue aims to **enhance the Bitcoin Price Prediction project** by introducing:

- A **Streamlit web application** for interactive prediction and visualization.
- Implementation of **Lasso Regression**, which was mentioned but not yet added.
- A fully detailed and structured **README.md** file for better usability and contributor onboarding.

These improvements will make the project more interactive, presentable, and open-source friendly.

---

### ✅ 1. Implement Lasso Regression
- [ ] Import and apply `Lasso` from `sklearn.linear_model`.
- [ ] Train the model using the existing dataset.
- [ ] Evaluate model performance using `r2_score` or similar metrics.
- [ ] Compare it with Linear, Ridge, and SVR models.

### ✅ 2. Build a Streamlit Web Application
- [ ] Build a UI with options to:
  - Upload CSV data (optional, or use existing dataset).
  - View data summary and EDA charts.
  - Select regression model (Linear, Ridge, Lasso, SVR).
  - Show prediction results and accuracy scores.
- [ ] Add performance metrics visualization (e.g., bar chart of model scores).
- [ ] Organize app code (e.g., `streamlit_app.py` or `app/` folder).

### ✅ 3. Improve `README.md` File
- [ ] Add project description and objective.
- [ ] Provide dataset source and sample link.
- [ ] List all models used with their performance.
- [ ] Add instructions:
  - Setup environment and install dependencies.
  - Run Jupyter notebook (EDA and training).
  - Launch the Streamlit web app.
- [ ] Include screenshots of the app (after completion).
- [ ] Add contribution guidelines for open source collaborators.

---

## 🗂 Recommended Folder Structure

```bash
bitcoin-price-prediction/
├── app/                 # Streamlit app code
├── notebooks/           # Jupyter notebooks (EDA + models)
├── models/              # Saved model files (if any)
├── data/                # Sample data or source links
├── README.md
├── requirements.txt
└── streamlit_app.py     # Web app entry point
```

## 🎯 Enhancement Aim

The aim of this enhancement is to **transform the Bitcoin Price Prediction project from a static notebook-based analysis into a dynamic, interactive, and user-friendly application**. By adding a **Streamlit web interface**, implementing the previously mentioned **Lasso Regression model**, and improving the **README.md** documentation, this upgrade will:

- Allow users to **interactively explore, visualize, and predict Bitcoin prices** without needing to run code manually.
- Enable better model performance evaluation by comparing multiple regressors (Linear, Ridge, Lasso, SVR).
- Make the project more accessible to **non-technical users** via a web interface.
- Attract **open-source contributors** with structured documentation and clear onboarding instructions.
- Lay the foundation for further improvements like deployment, CI/CD integration, or time-series models.

This enhancement will improve the project's usability, interactivity, and long-term maintainability as an open-source machine learning solution.

### Changes

✅ 1. Implement Lasso Regression
✅ 2. Build a Streamlit Web Application
✅ 3. Improve `README.md` File

### Guidelines

- [x] I have read the guidelines
- [ ] I have the link to my latest merged PR

### Full Name

Shivam Arora

### Participant Role

SSOC 4

**Comment by Shivu384** [comment]
@UTSAVS26 Please look at the issue I Raised !! also I would ask you to assign it under advance label as multiple tasks are to be performed.

**Comment by UTSAVS26** [comment]
@Shivu384 based on the work on the quality of the PR i will raise the difficulty

**Comment by UTSAVS26** [comment]
@Shivu384 I appreciate if there is any update on this issue by the end of tomorrow.

**Comment by Shivu384** [comment]
@UTSAVS26 Yes! I have updated the PR which i raised with no errors!! you can check
